clippy_lints/src/no_effect_replace.rs

```diff
@@ -0,0 +1,58 @@
+use clippy_utils::diagnostics::span_lint;
+use if_chain::if_chain;
+use rustc_ast::LitKind;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for `replace` statements which have no effect.
+    ///
+    /// ### Why is this bad?
+    /// It's either a mistake or confusing.
+    ///
+    /// ### Example
+    /// ```rust
+    /// "1234".replace("12", "22");
+    /// "1234".replacen("12", "22", 1);
```

Comment:
This wouldn't trigger the lint :smile:

```suggestion
    /// "1234".replace("12", "12");
    /// "1234".replacen("12", "12", 1);
```

Reply:
Right :D! Fixed